### PR TITLE
fix: calculates layout on desktop focus

### LINKED ISSUE #49: Can't create autostart layout
**Sub-command `once` didn't work at other desktop**
image

**Very interesting behaviour**
At desktop 1 i enter these commands:
```
[tw@arch-ng ~]$ bspc rule -a kitty:bspl desktop=6
[tw@arch-ng ~]$ bsp-layout set grid 6
[25695]
[tw@arch-ng ~]$ kitty --name bspl &
[1] 25724
[tw@arch-ng ~]$ kitty --name bspl &
[2] 25766
[tw@arch-ng ~]$ kitty --name bspl &
[3] 25808
[tw@arch-ng ~]$ kitty --name bspl &
[4] 25850
```
After that desktop 6 looks (no layout):
image
After that i create new node at desktop 6 and layout begins:
image
I think about events, so i create nodes at desktop 1 and layout listen for nodes at desktop 6, but can i create some layouts at boot via scripts with bsp-layout? If i create nodes first, `bsp-layout set` didn't organize them before new node appears, and `bsp-layout once` didn't work.

**Info**
Latest bsp-layout from AUR:
```
[tw@arch-ng ~]$ bsp-layout version

[tw@arch-ng ~]$
```
o_O

### PR DESCRIPTION
**Pull request by phenax**
Fixes #49, #44 and #12

### CODE REVIEW

**Comment by a2n-s** [comment]
Count on me to review the changes and test the fixes for the issues you mention :yum: 

i'll do that as soon as possible, i.e. after work :ok_hand:

**Comment by a2n-s** [comment]
i did verify that i was able to reproduce all the above issues on [master](https://github.com/phenax/bsp-layout) and i am

switching to [fix/issue-49-layout-on-desktop-focus](https://github.com/phenax/bsp-layout/tree/fix/issue-49-layout-on-desktop-focus), #44 appears to be gone   as i do not have the `No such file or directory` error when using `bsp-layout once` on an unfocused desktop :tada: 

however, when i:
  - put a few terminals on desktop `2`
  - run either `./build/bin/bsp-layout once {layout} 2` or `./build/bin/bsp-layout set {layout} 2` from desktop `4`

i still need to focus desktop `2` to apply the layout :thinking: 
and i've got a 2 monitor setup, so i can see that it does not apply directly :confused:

**Comment by a2n-s** [comment]
my bad, i miss understood the issue and the fix!

before the fix, `once` does not work at all on an unfocused desktop and `set` requires to switch to the desktop and produce either the `node_add`, the `node_remove` or the  `node_transfer` event from `bspc`

now, the layout applies as soon as the desktop has focus! :tada: 

that is a fix by my side :ok_hand:

File: src/layout.sh [hunk 1]
```diff
@@ -172,23 +172,43 @@ start_listener() {
...
+      desktop_name=$(get_desktop_name_from_id "$desktop_id");
+
+      if [[ "$desktop_name" = "$selected_desktop" ]]; then
+        __calculate_layout "$@";
+        exit 0
+      fi;
+    done & disown;
+  else
+    __calculate_layout "$@";
+  fi;
```
Comment (a2n-s) [inline src/layout.sh:211]: to be honest, i'm not comfortable enough with the code to comment a lot on it...

i do not see any obvious issue with that hunk and it solves 3 issues at the time :open_mouth: 

that is a go for me :ok_hand:

**Review by a2n-s — APPROVED** [review]
According to the knowledge i have on the code and given that 3 issues are solved at the same time, these changes can be merged to `master` :+1:

**Comment by phenax** [comment]
@a2n-s, fixed a minor issue, can you approve again?

**Review by a2n-s — APPROVED** [review]
Still fixes the issues :ok_hand:

**Pull request merged**